src/mbgl/gl/drawable_gl.cpp

```diff
@@ -245,27 +245,28 @@ gfx::StencilMode DrawableGL::makeStencilMode(PaintParameters& parameters) const
 }
 
 void DrawableGL::uploadTextures() const {
-    for (const auto& pair : textures) {
-        if (const auto& tex = pair.second) {
-            std::static_pointer_cast<gl::Texture2D>(tex)->upload();
+    for (const auto& texture : textures) {
+        if (texture) {
+            texture->upload();
         }
     }
 }
 
 void DrawableGL::bindTextures() const {
     int32_t unit = 0;
-    for (const auto& pair : textures) {
-        if (const auto& tex = pair.second) {
-            const auto& location = pair.first;
-            std::static_pointer_cast<gl::Texture2D>(tex)->bind(location, unit++);
+    for (size_t id = 0; id < textures.size(); id++) {
+        if (const auto& texture = textures[id]) {
+            if (const auto& location = shader->getSamplerLocation(id)) {
+                static_cast<gl::Texture2D&>(*texture).bind(static_cast<int32_t>(*location), unit++);
```

Comment:
Please verify if the cast is needed.

Reply:
Already verified. `bind` and `unbind` are platform specific. Probably that's why upload got casted, copy paste.

Reply:
https://github.com/maplibre/maplibre-native/blob/main/include/mbgl/gfx/texture2d.hpp